frontend/src/OnBoardingForm/OnbboardingFromSH.jsx

```diff
@@ -0,0 +1,304 @@
+                    src={
+                      darkMode
+                        ? 'assets/images/onboardingassets/Icons/Arrow_Back_dark.svg'
+                        : 'assets/images/onboardingassets/Icons/Arrow_Back.svg'
+                    }
+                    loading="lazy"
+                    alt="arrow back"
+                  />
+                  <p className="onboarding-back-text">Back</p>
+                </div>
+              )}
+              <div className="onboarding-bubbles-container">{page > 1 && <OnboardingBubblesSH page={page} />}</div>
+              {page > 1 && (
+                <div
+                  className="onboarding-back-button"
+                  onClick={() => {
+                    page != 4 && setPage((currPage) => currPage + 1);
+                    if (page == 4) {
+                      setIsLoading(true);
+                      setCompleted(true);
+                      return;
+                    }
+                  }}
+                >
+                  <p className="onboarding-skip-text">Skip</p>
+                  <img
+                    src={
+                      darkMode
+                        ? 'assets/images/onboardingassets/Icons/Arrow_forward_dark.svg'
+                        : 'assets/images/onboardingassets/Icons/Arrow_forward.svg'
+                    }
+                    loading="lazy"
+                    alt="arrow front"
+                  />
+                </div>
+              )}
+            </div>
+          </div>
+          <div className="form-container">
+            <div className="onboarding-header-wrapper">
+              <h1 className="onboarding-page-header">{FORM_TITLES[page]}</h1>
+              <p className="onboarding-page-sub-header">{FormSubTitles[0]}</p>
+            </div>
+            {page == 0 ? (
+              <AdminSetup {...pageProps} />
+            ) : page == 1 ? (
+              <WorkspaceSetupPage {...pageProps} />
+            ) : page == 2 ? (
+              <Page0 {...pageProps} setIsLoading={setIsLoading} />
+            ) : page == 3 ? (
+              <Page1 {...pageProps} setIsLoading={setIsLoading} />
+            ) : (
+              <Page2 {...pageProps} setIsLoading={setIsLoading} />
+            )}
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+// __PAGES__
+
+export function Page0({ formData, setFormData, setPage, page, setCompleted, isLoading, darkMode }) {
+  const props = { formData, setFormData, setPage, fieldType: 'companyName' };
+  const btnProps = {
+    setPage,
+    page,
+    formData,
+    setCompleted,
+    isLoading,
+    darkMode,
+  };
+  return (
+    <div className="onboarding-pages-wrapper">
+      <OnBoardingInput {...props} placeholder="Enter company name" autoFocus={true} />
+      <ContinueButtonSelfHost {...btnProps} />
+    </div>
+  );
+}
+
+export function Page1({ formData, setFormData, setPage, page, setCompleted, isLoading, darkMode }) {
+  const ON_BOARDING_ROLES = [
```

Comment:
If its the same for Self host and normal onboarding process you can reuse this as well